JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -143,53 +135,51 @@ private static string AggregateExceptionDetails(AggregateException ex)
             return innerExDetails.ToString();
         }
 
-        public void StopListening()
-        {
-            _cts.Cancel();
-            _log.LogInformation($"Stopping Listening - Queue: {_queue.QueueName}, Region: {_queue.Region.SystemName}");
-        }
-
         internal async Task ListenLoop(CancellationToken ct)
         {
             var queueName = _queue.QueueName;
             var region = _queue.Region.SystemName;
             ReceiveMessageResponse sqsMessageResponse = null;
 
-            try
+            while (!ct.IsCancellationRequested)
             {
-                sqsMessageResponse = await GetMessagesFromSqsQueue(ct, queueName, region).ConfigureAwait(false);
-                var messageCount = sqsMessageResponse.Messages.Count;
+                try
+                {
+                    sqsMessageResponse = await GetMessagesFromSqsQueue(ct, queueName, region).ConfigureAwait(false);
+                    var messageCount = sqsMessageResponse.Messages.Count;
 
-                _log.LogTrace($"Polled for messages - Queue: {queueName}, Region: {region}, MessageCount: {messageCount}");
-            }
-            catch (InvalidOperationException ex)
-            {
-                _log.LogTrace($"Could not determine number of messages to read from [{queueName}], region: [{region}]. Ex: {ex}");
-            }
-            catch (OperationCanceledException ex)
-            {
-                _log.LogTrace($"Suspected no message in queue [{queueName}], region: [{region}]. Ex: {ex}");
-            }
-            catch (Exception ex)
-            {
-                var msg = $"Issue receiving messages for queue {queueName}, region {region}";
-                _log.LogError(0, ex, msg);
-            }
+                    _log.LogTrace(
+                        $"Polled for messages - Queue: {queueName}, Region: {region}, MessageCount: {messageCount}");
+                }
+                catch (InvalidOperationException ex)
+                {
+                    _log.LogTrace(
+                        $"Could not determine number of messages to read from [{queueName}], region: [{region}]. Ex: {ex}");
+                }
+                catch (OperationCanceledException ex)
+                {
+                    _log.LogTrace($"Suspected no message in queue [{queueName}], region: [{region}]. Ex: {ex}");
```

Comment:
This could occur due to a timeout or because cancellation was requested by the calling code. Therefore this error message is misleading.

Reply:
There's a new `CancellationTokenSource` being created inside `GetMessagesFromSqsQueue` below, which is not being disposed.

Reply:
See [line 249](https://github.com/justeat/JustSaying/pull/403/files#diff-9005575960680ff0f5e43cfda505b5a2R249)

Reply:
:+1:, we should address that in separate PR IMO. It actually causes no issues, there was a leak with the linked token source leaking memory which was addressed. CTS is one of those things where most times Dispose is a no-op (see here for the full story: https://stackoverflow.com/a/30855490/2225808), but we shouldn't rely on this knowledge, we should do as the analyzer says here IMO.